Bug: likely m3d build error with ASSIMP_BUILD_NO_EXPORT

M3DImporter uses M3DWrapper:
https://github.com/assimp/assimp/blob/master/code/AssetLib/M3D/M3DImporter.cpp#L190
which is disabled with ASSIMP_BUILD_NO_EXPORT:
https://github.com/assimp/assimp/blob/master/code/AssetLib/M3D/M3DWrapper.h#L51
as well as required `m3d.h` inclusion.

```
M3DImporter.cpp: In member function 'virtual void Assimp::M3DImporter::InternReadFile(const string&, aiScene*, Assimp::IOSystem*)':
M3DImporter.cpp:190:20: error: variable 'Assimp::M3DWrapper m3d' has initializer but incomplete type
  190 |     M3DWrapper m3d(pIOHandler, buffer);
  ```
To clarify, it's not build system of assimp used, but issue looks quite apparent for me.
So if you enable M3D and disable exporters, build must fail.

--- a/code/AssetLib/M3D/M3DWrapper.cpp
+++ b/code/AssetLib/M3D/M3DWrapper.cpp
@@ -39,8 +39,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 ----------------------------------------------------------------------
 */
-#ifndef ASSIMP_BUILD_NO_M3D_IMPORTER
-#if !(ASSIMP_BUILD_NO_EXPORT || ASSIMP_BUILD_NO_M3D_EXPORTER)
+#if !defined ASSIMP_BUILD_NO_M3D_IMPORTER || !(defined ASSIMP_BUILD_NO_EXPORT || defined ASSIMP_BUILD_NO_M3D_EXPORTER)
 
 #include "M3DWrapper.h"
 
@@ -148,5 +147,4 @@ void M3DWrapper::ClearSave() {
 }
 } // namespace Assimp
 
-#endif
 #endif

--- a/code/AssetLib/M3D/M3DWrapper.h
+++ b/code/AssetLib/M3D/M3DWrapper.h
@@ -47,8 +47,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef AI_M3DWRAPPER_H_INC
 #define AI_M3DWRAPPER_H_INC
 
-#ifndef ASSIMP_BUILD_NO_M3D_IMPORTER
-#if !(ASSIMP_BUILD_NO_EXPORT || ASSIMP_BUILD_NO_M3D_EXPORTER)  
+#if !defined ASSIMP_BUILD_NO_M3D_IMPORTER || !(defined ASSIMP_BUILD_NO_EXPORT || defined ASSIMP_BUILD_NO_M3D_EXPORTER)
 
 #include <memory>
 #include <vector>
@@ -126,7 +125,6 @@ inline m3d_t *M3DWrapper::M3D() const {
 
 } // namespace Assimp
 
-#endif
 #endif // ASSIMP_BUILD_NO_M3D_IMPORTER
 
 #endif // AI_M3DWRAPPER_H_INC

